src/evm_parse.c

```diff
@@ -91,14 +133,12 @@ static void output_assetCall_prompt_to_string(char *const out, size_t const out_
   memcpy(&out[ix], to, sizeof(to));
   ix += sizeof(to) - 1;
 
-  out[ix] = '0'; ix++;
-  out[ix] = 'x'; ix++;
-  bin_to_hex_lc(&out[ix], out_size - ix, &in->address.val, ETHEREUM_ADDRESS_SIZE);
-  ix += 2 * ETHEREUM_ADDRESS_SIZE + 1;
+  output_evm_address_to_string(&out[ix], out_size - ix, in);
 }
 
-enum parse_rv parse_rlp_item(struct EVM_RLP_item_state *const state, evm_parser_meta_state_t *const meta);
-enum parse_rv parse_rlp_item_to_buffer(struct EVM_RLP_item_state *const state, evm_parser_meta_state_t *const meta);
+enum parse_rv parse_rlp_item(struct EVM_RLP_list_state *const state, evm_parser_meta_state_t *const meta);
```

Comment:
These look a little weird with the top-level EVM_RLP_list_state being passed in, but I do see that they go to the item's state internally and need to inspect a little bit more of the state.